### PR TITLE
Readme refactor

### PR DESCRIPTION
**Pull request by darrenvechain**
# Description

Refactor readme to give an overview, placing specific details in the `docs` directory.

Please refer to this link to see how it looks:

https://github.com/darrenvechain/thor/tree/readme-refactor

Once we're happy, I can update the links to use absolute links, rather than relative.

## Type of change

Please delete options that are not relevant.

- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,199 +1,68 @@
...
 
-This is the first implementation written in golang.
+## Getting Started
 
-[![Go](https://img.shields.io/badge/golang-%3E%3D1.19-orange.svg)](https://golang.org)
-[![Go Report Card](https://goreportcard.com/badge/github.com/vechain/thor)](https://goreportcard.com/report/github.com/vechain/thor)
-![GitHub Action Status](https://github.com/vechain/thor/actions/workflows/test.yaml/badge.svg)
-[![License](https://img.shields.io/badge/License-LGPL%20v3-blue.svg)](https://github.com/vechain/thor/blob/master/LICENSE)
-&nbsp;&nbsp; [![TG](https://img.shields.io/badge/chat-on%20telegram-blue)](https://t.me/VechainDevCommunity)
+VechainThor is the layer 1 blockchain, highly compatible† with Ethereum's ecosystem,which powers the vechain ecosystem.
```
Comment (leszek-vechain) [inline README.md:14]: do we need to have " ", after comma ? not sure what is proper format for that

File: README.md [hunk 2]
```diff
@@ -1,199 +1,68 @@
...
 
-To see a list of all available commands, run `make help`
+To chat with other community members you can join:
 
-### Building
+- [Telegram](https://t.me/VechainDevCommunity)
+- [Discord](https://t.me/VeChainDevCommunity)
 
-To build the main app `thor`, just run
+Do note that our [Code of Conduct](./docs/CODE_OF_CONDUCT.md) applies to all vecahin community channels. Users are *
```
Comment (leszek-vechain) [inline README.md:44]: typo in vecahin

File: docs/usage.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+```shell
+bin/thor -h
+```
+
+| Flag                        | Description                                                                                                |
+|-----------------------------|------------------------------------------------------------------------------------------------------------|
+| `--network`                 | The network to join (main\|test) or path to the genesis file                                               |
+| `--config-dir`              | The directory to use for common configuration files                                                        |
+| `--data-dir`                | Directory for blockchain databases (default: "/Users/darren/Library/Application Support/org.vechain.thor") |
```
Comment (leszek-vechain) [inline docs/usage.md:161]: maybe we can add {username} or something

File: docs/hosting-a-node.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+| CPU       | 8 Core                | 16 Core                   |
+| RAM       | 16 GB                 | 64 GB                     |
+| Bandwidth | 10 Mbit               | 20 Mbit                   |
+| Disk      | 500 GB SSD            | 2 TB SSD                  |
+
+___
+
+### Important Considerations
+
+_**Please note**: The recommendations and information below are based on the main network as of 22nd April 2024. The
```
Comment (leszek-vechain) [inline docs/hosting-a-node.md:50]: should we move it to top of page as the same is under system requirements ?

File: README.md [hunk 5]
```diff
@@ -1,199 +1,68 @@
...
-cd thor
-```
+The Vechain Improvement Proposals (VIPs) repository can be found [here](https://github.com/vechain/VIPs).
 
-To see a list of all available commands, run `make help`
+To chat with other community members you can join:
 
-### Building
+- [Telegram](https://t.me/VechainDevCommunity)
+- [Discord](https://t.me/VeChainDevCommunity)
```
Comment (leszek-vechain) [inline README.md:42]: discord and telegram links point to same location (telegram)

File: docs/usage.md (hunk 3 above)
Comment (darrenvechain) [reply docs/usage.md:161]: Yepp yepp, my bad

File: docs/installation.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+or build the full suite:
+
+```shell
+make all
+```
+
+If no errors are reported, all built executable binaries will appear in folder *bin*.
+
+```shell
+./bin/thor help
```
Comment (leszek-vechain) [inline docs/installation.md:35]: maybe we can add something - like run this to see help ?

File: docs/usage.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+To show usages of all command line options:
+
+```shell
+bin/thor -h
+```
+
+| Flag                        | Description                                                                                                |
+|-----------------------------|------------------------------------------------------------------------------------------------------------|
+| `--network`                 | The network to join (main\|test) or path to the genesis file                                               |
+| `--config-dir`              | The directory to use for common configuration files                                                        |
```
Comment (leszek-vechain) [inline docs/usage.md]: There is some small diffs in flags: 

```
VERSION:
   2.1.1-88c7c86d-release

COMMANDS:
     solo        client runs in solo mode for test & dev
     master-key  master key management
     help, h     Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --network value              the network to join (main|test) or path to genesis file
   --data-dir value             directory for block-chain databases (default: "/Users/leszek/Library/Application Support/org.vechain.thor")
   --cache value                megabytes of ram allocated to trie nodes cache (default: 4096)
   --beneficiary value          address for block rewards
   --target-gas-limit value     target block gas limit (adaptive if set to 0) (default: 0)
   --api-addr value             API service listening address (default: "localhost:8669")
   --api-cors value             comma separated list of domains from which to accept cross origin requests to API
   --api-timeout value          API request timeout value in milliseconds (default: 10000)
   --api-call-gas-limit value   limit contract call gas (default: 50000000)
   --api-backtrace-limit value  limit the distance between 'position' and best block for subscriptions APIs (default: 1000)
   --api-allow-custom-tracer    allow custom JS tracer to be used tracer API
   --verbosity value            log verbosity (0-9) (default: 3)
   --max-peers value            maximum number of P2P network peers (P2P network disabled if set to 0) (default: 25)
   --p2p-port value             P2P network listening port (default: 11235)
   --nat value                  port mapping mechanism (any|none|upnp|pmp|extip:<IP>) (default: "any")
   --bootnode value             comma separated list of bootnode IDs
   --skip-logs                  skip writing event|transfer logs (/logs API will be disabled)
   --pprof                      turn on go-pprof
   --disable-pruner             disable state pruner to keep all history
   --help, -h                   show help
   --version, -v                print the version
```

Comment (leszek-vechain) [reply docs/usage.md]: for e.g.: no config-dir

Comment (darrenvechain) [reply docs/usage.md]: Ya I'm not sure what that is. I took the flags from the codebase

Comment (leszek-vechain) [reply docs/usage.md]: ^ this is just run of ./bin/thor -h on main :), seems like some flags got hidden field set to true (not sure why)

**Review by otherview — DISMISSED** [review]
Minor stuff I think it would be nice to change but non-blockers.

Perhaps one interesting is shifting from Installation to Build. Installation doesn't feel very accurate.

File: README.md [hunk 8]
```diff
@@ -1,199 +1,68 @@
...
-[![Go Report Card](https://goreportcard.com/badge/github.com/vechain/thor)](https://goreportcard.com/report/github.com/vechain/thor)
-![GitHub Action Status](https://github.com/vechain/thor/actions/workflows/test.yaml/badge.svg)
-[![License](https://img.shields.io/badge/License-LGPL%20v3-blue.svg)](https://github.com/vechain/thor/blob/master/LICENSE)
-&nbsp;&nbsp; [![TG](https://img.shields.io/badge/chat-on%20telegram-blue)](https://t.me/VechainDevCommunity)
+VechainThor is the layer 1 blockchain, highly compatible† with Ethereum, which powers the vechain ecosystem.
+VechainThor is a public blockchain that is designed for mass adoption of blockchain technology by enterprise users of
+all sizes and is intended to serve as a foundation for a sustainable and scalable enterprise blockchain ecosystem.
 
-## Table of contents
+- † - VechainThor is currently up-to-date with the Ethereum's `Paris` hard fork, supporting Solidity version `0.8.18`.
```
[the remark this answers is not in the document]
Comment (darrenvechain) [reply README.md:18]: Its a reference to the description above:

```
... highly compatible† with Ethereum...
```

[the remark this answers is not in the document]
Comment (darrenvechain) [reply README.md:18]: image

**Review by otherview — APPROVED** [review]
lgtm 👍

**Pull request merged**